packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/endpoint_client/endpoint_info_store_manager.dart

```diff
@@ -1,34 +1,58 @@
-import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/endpoint_client/endpoint_store_keys.dart';
-import 'package:amplify_analytics_pinpoint_dart/src/impl/flutter_provider_interfaces/legacy_native_data_provider.dart';
+import 'package:amplify_analytics_pinpoint_dart/amplify_analytics_pinpoint_dart.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/endpoint_client/endpoint_global_fields_manager.dart';
 import 'package:amplify_core/amplify_core.dart';
 import 'package:amplify_secure_storage_dart/amplify_secure_storage_dart.dart';
 
-/// {@template amplify_analytics_pinpoint_dart.endpoint_info_store}
-/// Manages retrieval, storage, and management of Pinpoint Endpoint id
+/// Keys used to store info about the endpoint version store.
+enum EndpointStoreKey {
+  /// The current version of the endpoint version store
+  version,
+}
+
+/// The endpoint store version.
+enum EndpointStoreVersion {
+  /// The initial implementation of endpoint version store in Amplify Flutter Analytics.
+  v1,
+}
+
+/// Storage scope for Endpoint info
+const String endpointStorageScope = 'analyticsPinpoint';
+
+/// {@template amplify_analytics_pinpoint_dart.endpoint_store_manager}
+/// Manages retrieval, storage, and management of Pinpoint Endpoint id and fields
 /// {@endtemplate}
-class EndpointIdManager {
-  /// {@macro amplify_analytics_pinpoint_dart.endpoint_info_store}
-  EndpointIdManager({
-    required SecureStorageInterface store,
+class EndpointInfoStoreManager {
+  /// {@macro amplify_analytics_pinpoint_dart.endpoint_store_manager}
+  EndpointInfoStoreManager({
+    SecureStorageInterface? store,
     LegacyNativeDataProvider? legacyNativeDataProvider,
-    String? pinpointAppId,
-  })  : _store = store,
-        _legacyNativeDataProvider = legacyNativeDataProvider,
-        _pinpointAppId = pinpointAppId;
+  })  : _store = store ??
+            AmplifySecureStorageWorker(
+              config: AmplifySecureStorageConfig(
+                scope: endpointStorageScope,
+              ),
+            ),
+        _legacyNativeDataProvider = legacyNativeDataProvider;
 
   final SecureStorageInterface _store;
   final LegacyNativeDataProvider? _legacyNativeDataProvider;
-  final String? _pinpointAppId;
 
   static const String _endpointIdStorageKey = 'UniqueId';
 
+  /// Fields of the Pinpoint Endpoint
+  late final Future<EndpointGlobalFieldsManager> endpointFields = () async {
```

Comment:
Not sure I see the benefit of doing that. 

The behavior of EndpointGlobalFieldsManager is driven by the `SecureStorageInterface`.  The `SecureStorageInterface` should always be the same between EndpointGlobalFieldsManager and what is used to store the endpoint id.  

An important purpose of this class is to ensure that.  

Having EndpointGlobalFieldsManager as a DI suggests its functionality is separate from `SecureStorageInterface` which is not correct.  Consequently the user could give a EndpointGlobalFieldsManager initialized with another SecureStorageInterface than the one used in the EndpointInfoStoreManager which would cause endpoint info storage location to diverge. 

While I could refactor the EndpointGlobalFieldsManager to take a SecureStorageInterface in its init method, the user could still pass the EndpointGlobalFieldsManager DI initialized already.  And then if a second call to init changes that, that would revert what the user intended.